src/vehicle/message_parser/message_parser.go

```diff
@@ -82,28 +104,67 @@ func (parser *MessageParser) getKind(id uint16) (string, error) {
 		return "error", nil
 	}
 
+	if id == parser.infoId {
+		return "info", nil
+	}
+
 	parser.trace.Error().Uint16("id", id).Msg("unrecognized message id")
 	return "", fmt.Errorf("unrecognized message id")
 
 }
 
-func getProtectionContainer(kind string) (any, error) {
+func getProtection(kind string, payload []byte) (models.Protection, error) {
 	switch kind {
 	case "OUT_OF_BOUNDS":
-		return models.OutOfBounds{}, nil
+		var protection models.OutOfBounds
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "UPPER_BOUND":
-		return models.UpperBound{}, nil
+		var protection models.UpperBound
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "LOWER_BOUND":
-		return models.LowerBound{}, nil
+		var protection models.LowerBound
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "EQUALS":
-		return models.Equals{}, nil
+		var protection models.Equals
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "NOT_EQUALS":
-		return models.NotEquals{}, nil
+		var protection models.NotEquals
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "TIME_ACCUMULATION":
-		return models.TimeAccumulation{}, nil
+		var protection models.TimeLimit
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	case "ERROR_HANDLER":
-		return "", nil
+		var protection models.Error
+		err := json.Unmarshal(payload, &protection)
+		return models.Protection{
+			Kind: kind,
+			Data: protection,
+		}, err
 	default:
-		return nil, fmt.Errorf("protection kind not recognized: %s", kind)
+		return models.Protection{}, fmt.Errorf("protection kind not recognized: %s", kind)
 	}
 }
```

Comment:
create a separate function similar to this:

```go
func parseProtection[T any](kind string, payload []byte) (models.Protection, error) {
    var data T
    err := json.Unmarshal(payload, &data)
    return models.Protection {
        Kind: kind,
        Data: protection,
    }, err
}
```

and use it to avoid code duplication